src/App.ts

```diff
@@ -55,7 +55,7 @@ import {
   AppInitializationError,
   MultipleListenerError,
 } from './errors';
-import promiseAllsettled, { PromiseRejection } from 'promise.allsettled';
+import allSettled = require('promise.allsettled'); // tslint:disable-line:no-require-imports import-name
```

Comment:
i went down a massive rabbit hole trying to get this to work with regular ES Module `import ... from ...` syntax. I didn't leave a trail of where this led me in the code, but if anyone comes back looking for an explanation here it is:

there's an issue with a difference between the types and the runtime. its described in detail here: https://github.com/es-shims/Promise.allSettled/issues/5. the conclusion is that the rumtime behavior is spec-compliant, but typescript's output for compiling (downleveling) the import and calls are wrong. an issue was created for that: https://github.com/microsoft/TypeScript/issues/35420.

the [workaround suggested](https://github.com/es-shims/Promise.allSettled/issues/5#issuecomment-559389359) also did not work. a different error occurred (lost the exact error message but its not too hard to recreate if you need it). in the end i decided that the quickest fix here would be to just use the uglier, non-standard, less consistent `import` syntax.

hopefully when we update to a minimum node version of >= 12.9.0 or when we upgrade our minimum TypeScript version (which we don't declare right now) to one where the bug is fixed, this problem will just go away.